src/hotspot/cpu/riscv/assembler_riscv.hpp

```diff
@@ -266,12 +266,27 @@ class InternalAddress: public Address {
 class Assembler : public AbstractAssembler {
 public:
 
-  enum { instruction_size = 4 };
+  enum {
+    instruction_size = 4,
+    compressed_instruction_size = 2,
+  };
+
+  // instruction must start at passed address
+  static bool is_compressed_instr(address instr) {
+    if (UseRVC && (((uint16_t *)instr)[0] & 0b11) != 0b11) {
+      // 16-bit instructions end with 0b00, 0b01, and 0b10
```

Comment:
Looks like the comments is not correct here? We are checking the start instead of the end of the instruction encoding here. Suggestion: 
"16-bit instruction encoding starts with 0b00, 0b01, and 0b10"

Reply:
Thank you! The viewpoint might be different -- I was looking at the encoding graph, and it's bits `31 30 ... 3 2 1 0` so I used 'end with'; but 'start with' is also definitely correct because it's bits 0 and 1. So I checked the manual and was wondering if using its words ['lowest two bits'](https://github.com/riscv/riscv-isa-manual/blob/04cc07bccea63f6587371b6c75b228af3e5ebb02/src/intro.tex#L478-L482) might be more official?

Reply:
That also looks fine for me. Better to mention how the instruction is placemented in memory.

"Instructions are stored in memory as a sequence of 16-bit little-endian parcels, regardless of
memory system endianness. Parcels forming one instruction are stored at increasing halfword
addresses, with the lowest-addressed parcel holding the lowest-numbered bits in the instruction
specification."

Reply:
Thank you for the suggestion, Felix -- changed, and hope it looks good.

One another thing is the `instructions are stored in memory as a sequence of 16-bit little-endian parcels, regardless of memory system endianness` -- I was wondering if the current `Assembler::emit() -> Assembler::emit_int32()` could match it. Also when loading a 32-bit instruction from the memory - I think maybe the 16-bit parcels should be loaded, considering the 16-bit little-endian order, and be combined into one 32-bit instruction? - I cannot find a big-endian simulator to test that, and feel glad to receive any input.

Reply:
I don't quite understand what you mean here. But let's discuss that as a separate issue when you have more details.

Reply:
It seems I failed to say it clear, but I also feel glad putting it afterward, for it might have nothing to do with this patch.